src/test/fuzz/util/net.cpp

```diff
@@ -21,6 +21,7 @@
 #include <cstdlib>
 #include <cstring>
 #include <thread>
+#include <unordered_map>
```

Comment:
Looks wrong as well?

Reply:
Well, in `FuzzedSock::WaitMany()` function https://github.com/bitcoin/bitcoin/blob/4717a5aa31bc9cdb5406f4161f4c600e5c2659c0/src/test/fuzz/util/net.cpp#L332-L339 the type `EventsPerSock` is actually an `std::unordered_map`: https://github.com/bitcoin/bitcoin/blob/2ec97825e70882262ee2b7d53dc1def788a60162/src/util/sock.h#L218

Reply:
Right, so it would need to include `util/sock.h` instead to get `EventsPerSock`?

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/26763#issuecomment-1370913829).

Some relevant IWYU's docs: https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/WhatIsAUse.md#re-exporting.

Reply:
This will force iwyu to remove `#include <unordered_map>` if `util/sock.h` is included, but an unrelated code piece needs the unordered map include?

Not sure which is preferable. I am just explaining the tradeoffs.

Reply:
What about `// IWYU pragma: no_include <unordered_map>` in `src/test/fuzz/util/net.cpp`?

Reply:
Seems worse (combining all downsides), compared to just including the file

Reply:
FWIW, I lean to just following IWYU's suggestion and including a header. It won't make a translation unit bigger anyway.